src/commands/mobile_app/upload.rs

```diff
@@ -198,19 +198,106 @@ fn validate_is_mobile_app(path: &Path, bytes: &[u8]) -> Result<()> {
             debug!("Detected APK file");
             return Ok(());
         }
+
+        if is_ipa_file(bytes)? {
+            debug!("Detected IPA file");
+            return Ok(());
+        }
     }
 
     debug!("File format validation failed");
     Err(anyhow!(
-        "File is not a recognized mobile app format (APK, AAB, or XCArchive): {}",
+        "File is not a recognized mobile app format (APK, AAB, XCArchive, or IPA): {}",
         path.display()
     ))
 }
 
+fn ipa_to_xcarchive(ipa_path: &Path, ipa_bytes: &[u8]) -> Result<TempFile> {
+    debug!(
+        "Converting IPA to XCArchive structure: {}",
+        ipa_path.display()
+    );
+
+    let temp_dir = crate::utils::fs::TempDir::create()?;
+    let xcarchive_dir = temp_dir.path().join("archive.xcarchive");
+    let products_dir = xcarchive_dir.join("Products");
+    let applications_dir = products_dir.join("Applications");
+
+    debug!("Creating XCArchive directory structure");
+    std::fs::create_dir_all(&applications_dir)?;
+
+    // Extract IPA file
+    let cursor = std::io::Cursor::new(ipa_bytes);
+    let mut ipa_archive = zip::ZipArchive::new(cursor)?;
+
+    let mut app_name = String::new();
+
+    // Extract .app from Payload/ directory
+    for i in 0..ipa_archive.len() {
+        let mut file = ipa_archive.by_index(i)?;
+        let file_path = file.name().to_string();
```

Comment:
[change request]

Using `ZipFile::name` directly [is dangerous](https://docs.rs/zip/latest/zip/read/struct.ZipFile.html#warnings), as it could break out of the current directory (e.g. if the path is `Payload/../../../something`). We could then end up overwriting this path later.

Please use [`ZipFile::enclosed_name`](https://docs.rs/zip/latest/zip/read/struct.ZipFile.html#method.enclosed_name) instead, as the enclosed name is validated to ensure it does not break out of the current directory.

Reply:
As a side note, we should probably make sure we don't use `ZipFile::name` anywhere else in the codebase.